pkg/agent/route/route_linux.go

```diff
@@ -409,7 +410,8 @@ func (c *Client) syncIPSet() error {
 		})
 	}
 
-	if c.connectUplinkToBridge {
+	// FlexibleIPAM is available in encap/noEncap/hybrid mode.
+	if c.connectUplinkToBridge && !c.networkConfig.TrafficEncapMode.IsNetworkPolicyOnly() {
```

Comment:
FlexibleIPAM is only available in noEncap mode, the check is not needed.

Reply:
Got that. I found that there is a validation for traffic mode when Flexible IPAM is enabled, so the check is redundant. I will remove the check.